src/core/game.js

```diff
@@ -0,0 +1,99 @@
+import Coords from './coords';
+
+export const EMPTY_VALUE = 0;
+
+export const buildGrid = (size = 4) => {
+    let value = EMPTY_VALUE;
+    let grid = [];
+
+    for (let y = 0; y < size; y++) {
+        grid[y] = [];
+
+        for (let x = 0; x < size; x++) {
+            value++;
+            grid[y][x] =
+                value == size * size ? EMPTY_VALUE : (grid[y][x] = value);
+        }
+    }
+    return grid;
+};
+
+export const deepCopyGrid = grid => grid.map(row => row.slice());
```

Comment:
It should work but it's more common to see: `cloned = [...source]` for array cloning.